src/shared/utilities/tsUtils.ts

```diff
@@ -4,7 +4,13 @@
  */
 
 export const getPropAs = <T>(obj: any, key: string) => {
-    return ((obj as any) as {
-        [key: string]: T
-    })[key]
+    return (
+        obj as any as {
+            [key: string]: T
+        }
+    )[key]
 }
+
+type NoSymbols<T> = { [Property in keyof T]: Property extends symbol ? never : Property }[keyof T]
+export type InterfaceNoSymbol<T> = Pick<T, NoSymbols<T>>
+export type ClassToInterface<T> = Pick<T, keyof T>
```

Comment:
This is inaccurate: this should be named `ClassToType` since this is no longer an interface.

Reply:
They are effectively the same thing. The only different is types cannot be re-declared for extension vs. interfaces. TS types are purely structural so externally they look the same.

Reply:
It could be confusing, but I tend to agree that semantically "interface" is more descriptive of the purpose, and also more discoverable. Confusion can be avoided with a tactical docstring:
```suggestion
/**
 * Defines a type from the public keys of class T. 
 *
 * Note: TypeScript types are purely structural so externally the result looks the same as
 * one declared with the `interface` keyword. The only difference from a literal `interface`
 * is that it cannot be re-declared for extension.  
 */
export type ClassToInterface<T> = Pick<T, keyof T>
```